ts_src/transaction_builder.ts

```diff
@@ -900,6 +994,43 @@ function build(
 
       return payments.p2pk({ signature: signatures[0] });
     }
+    case SCRIPT_TYPES.P2AR: {
+      const m = input.vaultTxType === VaultTxType.Alert ? 1 : 2;
```

Comment:
I think that it would give more information what is going on here
`const m = input.vaultTxType === VaultTxType.Alert ? VaultTxType.Alert : VaultTxType.Recovery;` 

Or maybe we can validate the vaultTxType and throw an error if it is not a VaultTxType.Alert or VaultTxType.Recovery for preventing the user from passing unintentionally wrong vaultTxType?

Reply:
nope, those numbers are not derived from txType enum. **m** number defines how many signatures are required

Reply:
export enum VaultTxType {
  Alert, // 0
  Instant, // 1
  Recovery, // 2
  NonVault, // 3
}